File: qc-application/src/main/java/com/eager/questioncloud/application/api/workspace/service/CreatorRegisterService.java
package com.eager.questioncloud.application.api.workspace.service;

import com.eager.questioncloud.application.api.workspace.implement.CreatorRegister;
import com.eager.questioncloud.core.domain.creator.implement.CreatorStatisticsProcessor;
import com.eager.questioncloud.core.domain.creator.model.Creator;
import com.eager.questioncloud.core.domain.creator.model.CreatorProfile;
import com.eager.questioncloud.core.domain.user.model.User;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class CreatorRegisterService {
    private final CreatorRegister creatorRegister;
    private final CreatorStatisticsProcessor creatorStatisticsProcessor;

    public Creator register(User user, CreatorProfile creatorProfile) {
        Creator creator = creatorRegister.register(user, creatorProfile);
        creatorStatisticsProcessor.initializeCreatorStatistics(creator.getId());
        return creator;
    }
}

File: qc-application/src/main/java/com/eager/questioncloud/application/api/creator/service/CreatorRegisterService.kt
package com.eager.questioncloud.application.api.creator.service

import com.eager.questioncloud.application.api.creator.implement.CreatorRegister
import com.eager.questioncloud.application.api.creator.implement.CreatorStatisticsInitializer
import com.eager.questioncloud.core.domain.creator.model.Creator
import com.eager.questioncloud.core.domain.question.enums.Subject
import com.eager.questioncloud.core.domain.user.model.User
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class CreatorRegisterService(
    private val creatorRegister: CreatorRegister,
    private val creatorStatisticsInitializer: CreatorStatisticsInitializer,
) {
    @Transactional
    fun register(user: User, mainSubject: Subject, introduction: String): Creator {
        val creator = creatorRegister.register(user, mainSubject, introduction)
        creatorStatisticsInitializer.initializeCreatorStatistics(creator.id!!)
        return creator
    }
}
